src/shims/env.rs

```diff
@@ -0,0 +1,61 @@
+use std::collections::HashMap;
+
+use rustc::ty::{layout::{Size, Align}, TyCtxt};
+use rustc_mir::interpret::{Pointer, Memory};
+use crate::stacked_borrows::Tag;
+use crate::*;
+
+#[derive(Default)]
+pub struct EnvVars {
+    map: HashMap<Vec<u8>, Pointer<Tag>>,
+}
+
+impl EnvVars {
+    pub(crate) fn init<'mir, 'tcx>(
+        communicate: bool,
+        ecx: &mut InterpCx<'mir, 'tcx, Evaluator<'tcx>>,
+        tcx: &TyCtxt<'tcx>,
```

Comment:
You can use `ecx.tcx.tcx` for the `TyCtxt`.

Reply:
Ah, good point. And please move the `ecx` first, it is basically "self" here.

Reply:
I think I cannot, becuase ecx is shared mutably to get `Memory` then calling `&ecx.tcx.tcx` would create a second reference.

Reply:
Do `let tcx = {ecx.tcx.tcx};` as first line in the body, that does a *copy* of the shared ref instead of a reborrow so you can still use `ecx`.

Reply:
Ohhh, good to know.